src/carma-platform/carma_system_resource_usage_plot.py

```diff
@@ -0,0 +1,312 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+from datetime import datetime
+import argparse
+import sys
+import os
+from guidance_scripts import *
+
+ENVIRONMENT_MAP_UPDATE_TOPIC = "/environment/map_update"
+ENVIRONMENT_SEMANTIC_MAP_TOPIC = "/environment/semantic_map"
+
+
+def calculate_time_offset(mcap_path, csv_file):
+    """
+    Calculate the time offset between MCAP and CSV data by comparing their hours
+
+    Args:
+        mcap_path (str): Path to the MCAP file
+        csv_file (str): Path to the CSV file
+
+    Returns:
+        timedelta: Time offset between MCAP and CSV data
+    """
+    # Get MCAP start time
+    reader, type_map, global_start_time = open_bagfile(str(mcap_path))
+    mcap_start_time = pd.to_datetime(global_start_time)
+
+    # Get CSV start time
+    df = pd.read_csv(csv_file)
+    df["Timestamp"] = pd.to_datetime(df["Timestamp"])
+    csv_start_time = df["Timestamp"].min()
+
+    # Calculate hour difference
+    hour_diff = mcap_start_time.hour - csv_start_time.hour
+    time_offset = pd.Timedelta(hours=hour_diff)
+
+    print(f"MCAP start time: {mcap_start_time}")
+    print(f"CSV start time: {csv_start_time}")
+    print(f"Hour difference: {hour_diff}")
+    print(f"Time offset: {time_offset}")
```

Comment:
Are hour difference and time offset the same since `time_offset` is created just using `hour_diff`

Reply:
removed hour_diff